### DISCUSSION TITLE
Child Theme / Managing updates

### DISCUSSION THREAD

**Discussion by moggiex** [colbyfayock/next-wordpress-starter#452] [General]
Howdy [colby](https://github.com/colbyfayock), All,

Daft question... And this is probably due to my lack of understanding of next.js. 

If I clone this repo, but then make 'structural' edits to say [src/pages/500.js](https://github.com/colbyfayock/next-wordpress-starter/blob/main/src/pages/500.js) if this page is updated in the repo, managing the update to this page and others over time as the original repo changes and the customised site also changes would turn into a nightmare.

How can this be managed? The best reference is child themes in WordPress, which overrides the original theme (in this case a page or maybe a component).

Matt

**Comment by colbyfayock** [comment]
@moggiex are you basically saying how can you create a custom website based off of this template, with desired changes, that can easily inherit updates to the template project?

**Reply by moggiex** [reply]
Yes!

**Reply by colbyfayock** [reply]
im not sure there's going to be a _great_ way to handle this type of thing

from first thought the theme would need to be abstracted into a library, where all of the functionality is happening through helper functions / wrappers instead of rendered out with react components, though the library could even include some default templates that could wrap the application

just spitballing, would be interesting, but i'm not sure that i have the time to invest in that with other priorities at the moment

**Reply by moggiex** [reply]
Appreciate its not going to be simple from what you've mentioned. Just makes it unrealistic to use and chase through updates later with a heavily modified theme etc... 👍

**Reply by colbyfayock** [reply]
i think that's generally the standard practice of most "Starters" in the React world

i've seen some interesting examples i/we could take inspiration from, but that was never the point of the design

https://github.com/shuding/nextra
